File: book_app/lib/apps/models/book_model.dart
class BookModel {
    String? id;
    String? title;
    String? description;
    double? price;
    List<String>? images;

    BookModel({
        this.id,
        this.title,
        this.description,
        this.price,
        this.images,
    });

    factory BookModel.fromJson(Map<String, dynamic> json) => BookModel(
        id: json["_id"],
        title: json["title"],
        description: json["description"],
        price: json["price"]?.toDouble(),
        images: json["images"] == null ? [] : List<String>.from(json["images"]!.map((x) => x)),
    );

    Map<String, dynamic> toJson() => {
        "_id": id,
        "title": title,
        "description": description,
        "price": price,
        "images": images == null ? [] : List<dynamic>.from(images!.map((x) => x)),
    };
}

File: bookstore/src/models/book_model.js
const { Schema, model } = require("mongoose");

const bookSchema = new Schema({
  title: { type: String, required: [true, "title is required"] },
  description: { type: String, default: "" },
  price: { type: Number, default: "" },
  images: { type: Array, default: [] },
  updatedOn: { type: Date },
  createdOn: { type: Date },
});

bookSchema.pre("save", function (next) {
  this.updatedOn = new Date();
  this.createdOn = new Date();

  next();
});

bookSchema.pre(["update", "findOneAndUpdate", "updateOne"], function (next) {
  const update = this.getUpdate();
  delete update._id;

  this.updatedOn = new Date();

  next();
});

const BookModel = model("Book", bookSchema);

module.exports = BookModel;
